libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/ReportBatchBuilder.java

```diff
@@ -13,42 +13,39 @@
     // Once this size is reached, data is persisted to disk, mCurrentReports is cleared.
     public static final int MAX_REPORTS_IN_MEMORY = 50;
     private static final String LOG_TAG = LoggerUtil.makeLogTag(ReportBatchBuilder.class);
-    private final ArrayList<String> reports = new ArrayList<String>();
```

Comment:
Woot.  This ought to kill all those `finalizeReports()` memory crashes we see on lower end devices like the Nexus S.